cpp/include/kvikio/defaults.hpp

```diff
@@ -163,19 +205,46 @@ class defaults {
    *  - when `/run/udev` isn't readable, which typically happens when running inside a docker
    *    image not launched with `--volume /run/udev:/run/udev:ro`
    *
-   * @return The boolean answer
+   * @return Compatibility mode.
    */
-  [[nodiscard]] static bool compat_mode() { return instance()->_compat_mode; }
+  [[nodiscard]] static CompatMode compat_mode()
```

Comment:
we can still return bool here

Reply:
from an offline discussion, it looks like we need to avoid reducing the compat mode state to bool. Resolving.